app.go

```diff
@@ -224,7 +232,7 @@ func (a *App) Run(arguments []string) (err error) {
 
 	err = parseIter(set, a, arguments[1:], shellComplete)
 	nerr := normalizeFlags(a.Flags, set)
-	context := NewContext(a, set, nil)
+	context := NewContext(a, set, &Context{Context: ctx})
```

Comment:
Will this still work of user does RunWithContext(nil, args)

Reply:
Passing a nil Context is always a developer error.

[From the docs](https://golang.org/pkg/context/):

> Do not pass a nil Context, even if a function permits it. Pass context.TODO if you are unsure about which Context to use. 

We shouldn't need to support that